lib/charms/layer/openstack.py

```diff
@@ -130,14 +130,23 @@ def detect_octavia():
     """
     Determine whether the underlying OpenStack is using Octavia or not.
 
-    Returns True if Octavia is found, and False otherwise.
+    Returns True if Octavia is found in the region, and False otherwise.
     """
     try:
-        catalog = {s['Name'] for s in _openstack('catalog', 'list')}
+        creds = _load_creds()
+        region = creds['region']
+        octavia_found = False
+        for catalog in _openstack('catalog', 'list'):
+            if catalog['Name'] == 'octavia':
+                if catalog.get('Endpoints'):
+                    if catalog['Endpoints'][0]['region'] == region:
```

Comment:
This is 3 nested if statements instead of one with two `and` combinators.

Reply:
Thought about all the review comments before submitting initial PR and made wrong silly choices. Thanks for the review.
This is resolved in new patch.